packages/common/src/extensions/slickCellExternalCopyManager.ts

```diff
@@ -127,15 +127,15 @@ export class SlickCellExternalCopyManager {
 
     // if a custom getter is not defined, we call serializeValue of the editor to serialize
     if (columnDef) {
-      if (columnDef.editor) {
+      if (columnDef.editor?.model) {
```

Comment:
actually I think it would be best to call the one that the SlickGrid is aware of which is `editorClass` since SlickGrid doesn't have any knowledge whatsoever of `editor.model` (you won't find `editor.model` anywhere in SlickGrid core file, it is filled later by the wrapper which can be VanillaGridBundle, AngularSlickgrid, AureliaSlickgrid, ...). Note that there a few of them to replace

Reply:
thats interesting bc the examples all assign the model directly in the column definitions

hmm that also means though, while internalEditor is only deprecated but not gone we should take that first before editorClass?

Reply:
yeah I understand that it might be confusing, it might be just in my head but because I want to keep my `slickgrid.core.ts` file as close as possible to the original SlickGrid project. Some properties are not meant to be used within SlickGrid core file, for example any props related to export services and any plugin props and so on.. when I wasn't sure if I was going to still use `6pac/slickgrid` in here, I added Generics on SlickGrid that could be extended and so when I was using it in the wrapper, I could simply provide my custom `GridOption` and `Column` interfaces from the wrapper and SlickGrid was happy with it but it's just an extra layer that I would rather not really use too much (in fact I still have a lot of places where I have `grid: SlickGrid` instead of the full typed ref of `grid: SlickGrid<GridOption, Column>` because I find it too long and I don't have to do that anymore since I merged SlickGrid in here and typically the props I want to use are already in the GridOption/Column base interfaces)... so long story short, it's kind of "only me" knowing this little thing but I'm not sure I really want to go with the full Generics+extend interface way either because I think it would overcomplicate things.

https://github.com/ghiscoding/slickgrid-universal/blob/abe772b9ad9480688ef000dddfa86d2cdc6b7e52/packages/common/src/core/slickGrid.ts#L123-L124

Reply:
> hmm that also means though, while internalEditor is only deprecated but not gone we should take that first before editorClass?

the `internalEditor` will be removed completely in the next major and its goal was simply to hide the complexity of Editor creation to the user. I didn't want to introduce a breaking change so it's still in place and still work but will be removed completely in next major. SlickGrid always had a `editor` grid option (it never had the `filter` option though) and it was used like this `editor: Slick.longText`, but since I use a few external libraries (ms-select, autocomplete, flatpickr, ...), I needed a way to potentially provide extra options and I strongly felt that the developer would still prefer to use `editor` without necessarily caring about how it was used internally which is why at the end I came up with that `internalEditor` thing, it's something that was meant to be used only within the wrapper and hidden to the user...but in some cases the user want to have access to the editor instance and modify it (e.g. add some items to the ms-select `collection`) and in that case the user needed to reference the instance (not the class itself that SlickGrid editor had) and that was confusing that the editor was creating it with `editor` but later had to reference it with `internalEditor`... that's long story of how it all started, hope it helps understanding why I'm trying to make it simpler ;)

Reply:
ok got it. yeah I make use of the internalEditor a couple of times in my ForeignKey Filter/Editor so I understand what you're trying to simplify here.